orchestra/tests/test_views.py

```diff
@@ -131,3 +150,81 @@ def test_time_entries_post_data_invalid(self, mock_func, mock_logger):
                                         data=json.dumps(self.time_entry_data),
                                         content_type='application/json')
         self._verify_bad_request(resp, "['Error']")
+
+
+class TimerEndpointTests(EndpointTests):
+
+    def setUp(self):
+        super().setUp()
+        user = UserFactory(username='test_user')
+        self.worker = WorkerFactory(user=user)
+        self.request_client = RequestClient()
+        self.request_client.login(username=self.worker.user.username,
+                                  password='defaultpassword')
+        self.time = timezone.now()
+
+    @patch('orchestra.views.time_tracking.start_timer')
+    def test_start_timer(self, mock_start):
+        timer = TaskTimer(worker=self.worker, start_time=self.time)
+        timer.save()
+        mock_start.return_value = timer
+        resp = self.request_client.post(
+            reverse('orchestra:orchestra:start_timer'),
+            data=json.dumps({'assignment': '11'}),
+            content_type='application/json')
+        mock_start.assert_called_with(self.worker, assignment_id='11')
+        data = json.loads(resp.content.decode())
+        serializer = TaskTimerSerializer(timer)
+        self.assertEqual(data, serializer.data)
+
+    @patch('orchestra.views.time_tracking.start_timer',
+           side_effect=TaskAssignment.DoesNotExist)
+    def test_start_timer_worker_not_assigned(self, mock_start):
+        resp = self.request_client.post(
+            reverse('orchestra:orchestra:start_timer'),
+            data=json.dumps({'assignment': '11'}),
+            content_type='application/json')
+        self._verify_bad_request(resp,
+                                 'Worker is not assigned to this task id.')
+
+    @patch('orchestra.views.time_tracking.start_timer',
+           side_effect=TimerError('test'))
+    def test_start_timer_timer_already_running(self, mock_start):
```

Comment:
i'm testing how the endpoint handles when the mocked function throws a `TimerError` exception. TimerEndpointTests is only testing the logic in views.py, not the logic in utils.time_tracking. That code is tested in test_time_tracking